src/boost_histogram/_internal/axis.py

```diff
@@ -444,24 +454,22 @@ def __init__(self, start, stop, **kwargs):
         # underflow and overflow settings are ignored, integers are always
         # finite and thus cannot end up in a flow bin when growth is on
         if "growth" in options and "circular" not in options:
-            self._ax = ca.integer_growth(start, stop, metadata)
+            self._ax = ca.integer_growth(start, stop)
         elif options == {"underflow", "overflow"}:
-            self._ax = ca.integer_uoflow(start, stop, metadata)
+            self._ax = ca.integer_uoflow(start, stop)
         elif options == {"underflow"}:
-            self._ax = ca.integer_uflow(start, stop, metadata)
+            self._ax = ca.integer_uflow(start, stop)
         elif options == {"overflow"}:
-            self._ax = ca.integer_oflow(start, stop, metadata)
-        elif (
-            "circular" in options and "growth" not in options
-        ):  # growth=True should work
-            self._ax = ca.integer_circular(
-                start, stop, metadata
-            )  # flow bins do no matter
+            self._ax = ca.integer_oflow(start, stop)
+        elif "circular" in options and "growth" not in options:
+            self._ax = ca.integer_circular(start, stop)
         elif options == set():
-            self._ax = ca.integer_none(start, stop, metadata)
+            self._ax = ca.integer_none(start, stop)
         else:
             raise KeyError("Unsupported collection of options")
 
+        self.metadata = metadata
```

Comment:
It looks like you are not passing metadata to the cpp instance anymore. Is that correct? If cpp code cannot see metadata, it cannot pass it on in methods like `.project` or the wrapper over `reduce`. `operator==` also compare metadata.

Reply:
In am not against a solution that handles metadata completely on the Python side and removes metadata_t in C++, but then the python interface over project and reduce, operator== etc need to handle metadata explicitly (before it was handled automatically by the underlying cpp implementation).

Reply:
That would be a large, heavy rewrite, and would replicate a lot of the logic for computing axes in boost-histogram, rather than just letting Boost.Histogram do the work. I'm not against it either, but I like the simplicity of using Boost.Histogram's computation (since it can handle this) instead of replicating logic.

Reply:
Ok, no need to rewrite things if things are working as they are now. Apparently I misunderstood.

Context: What I like about cpp not seeing the Python metadata is the clean border between the C++ world and the Python world. Life-time issues and (deep) copy issues are are more easily handled in this way.